packages/core/src/routes/connector.ts

```diff
@@ -0,0 +1,76 @@
+import { Connectors } from '@logto/schemas';
+import { object, string } from 'zod';
+
+import { getConnectorInstanceById } from '@/connectors';
+import { getAllConnectorInfo, getConnectorInfoById } from '@/lib/connectors';
+import koaGuard from '@/middleware/koa-guard';
+import { findConnectorById, updateConnector } from '@/queries/connector';
+
+import { AuthedRouter } from './types';
+
+export default function connectorRoutes<T extends AuthedRouter>(router: T) {
+  router.get('/connectors', async (ctx, next) => {
+    ctx.body = await getAllConnectorInfo();
+
+    return next();
+  });
+
+  router.get(
+    '/connectors/:id',
+    koaGuard({ params: object({ id: string().min(1) }) }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+      } = ctx.guard;
+      ctx.body = await getConnectorInfoById(id);
+
+      return next();
+    }
+  );
+
+  router.patch(
+    '/connectors/:id/enabled',
+    koaGuard({
+      params: object({ id: string().min(1) }),
+      body: Connectors.createGuard.pick({ enabled: true }),
+    }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+        body: { enabled },
+      } = ctx.guard;
+      await findConnectorById(id);
+      await updateConnector({ set: { enabled }, where: { id } });
+      ctx.body = { enabled };
+
+      return next();
+    }
+  );
+
+  router.patch(
+    '/connectors/:id',
+    koaGuard({
+      params: object({ id: string().min(1) }),
+      body: Connectors.createGuard
+        .omit({ id: true, type: true, enabled: true, createdAt: true })
+        .partial(),
+    }),
+    async (ctx, next) => {
+      const {
+        params: { id },
+        body,
+      } = ctx.guard;
+      await findConnectorById(id);
+      const connectorInstance = getConnectorInstanceById(id);
```

Comment:
we don't query db here since we're using `getConnectorById` in `lib/` as the SSOT